### ISSUE TITLE
Conducting REGENIE with the X chromosome leads to error

### ISSUE THREAD

**Issue by Sabor117** [rgcgithub/regenie#451]
This is a [duplication of a comment I made in another thread](https://github.com/rgcgithub/regenie/issues/430), which also was related to the same issue I am about to describe. However, that one had been previously closed so I've instead opened a new this one. Any and all help appreciated!

I have recently conducted a full genome run of REGENIE in UK Biobank across chromosomes 1-22 with absolutely no issue, following the REGENIE UKB tutorial (I.e. made a full bed file of all chromosomes using the Plink command described, and then running step 1 and step 2 as in the tutorial). However we want to run REGENIE on the X chromosome as well if possible, so I have gone back to the start to re-do the analysis.

I again created a full bed/bim/fam file set of all chromosomes, now including the X chromosome (which was seemingly automatically encoded as chromosome 23 by plink). Then when I ran step 1 on this new bed file it threw the following error:

```
making predictions...regenie_v3.2.8.gz_x86_64_Centos7_mkl: /src/regenie/external_libs/eigen-3.4.0/Eigen/src/Core/DenseCoeffsBase.h:427: Eigen::DenseCoeffsBase<Derived, 1>::Scalar& Eigen::DenseCoeffsBase<Derived, 1>::operator()(Eigen::Index) [with Derived = Eigen::Array<int, -1, 1>; Eigen::DenseCoeffsBase<Derived, 1>::Scalar = int; Eigen::Index = long int]: Assertion index >= 0 && index < size()' failed.
```

Now I had noted in the tutorial that:

> To include X chromosome genotypes in step 1 and/or step 2, males should be coded as diploid so that their genotypes are 0/2

But also that:

> this is done automatically for BED file formats

However, I would like to make absolutely certain that this is the case and if not ask whether I should in fact be changing the plink command (entered below) in order to ensure that the BEDs are encoded as diploid correctly.

```
plink \
  --bed ukb_cal_chr1_v2.bed \
  --bim ukb_snp_chr1_v2.bim \
  --fam ukbXXX_int_chr1_v2_s488373.fam \
  --merge-list list_beds.txt \
  --make-bed --out ukb_cal_allChrs
```

Assuming they ARE being coded correctly/automatically by Plink, I would then like to ask if there is anything else I need to check/fix to try and sort out this error? I am in the process of re-running a test Step 1 with the `--debug` and `--verbose` arguments to try and get a better handle on what is causing the problem and will come back when I get the output for that.

**Comment by Sabor117** [comment]
Having now re-run with --debug and --verbose, the output log file actually seems to be absolutely no different... So I'm not sure how to solve this.

This was the command I ran (directories edited):

```
Options in effect:
  --step 1 \
  --bed genotypes/full_plink/ukb22418_allChrs \
  --phenoFile FINAL_hes_based_phenotype_frame_REGENIE.tsv \
  --covarFile gwas_cost_covariates_population_EUR.tsv \
  --extract /plink_superpop_qc/EUR_qc_snps_for_regenie.snplist \
  --keep /plink_superpop_qc/EUR_qc_snps_for_regenie.id \
  --qt \
  --debug \
  --verbose \
  --bsize 1000 \
  --lowmem \
  --lowmem-prefix /tmpdir/regenie_tmp_preds/ \
  --out /regenie/step1_output/
```

**Comment by joellembatchou** [comment]
Hi,

This issue was fixed in [v3.2.9](https://github.com/rgcgithub/regenie/releases/tag/v3.2.9).

Cheers,
Joelle

**Comment by Sabor117** [comment]
I can confirm that this did indeed work when upgrading to the new version!

Thanks for getting back to me about this Joelle!
